torch/_dynamo/variables/builder.py

```diff
@@ -1724,6 +1728,28 @@ def wrap_tensor(self, value: torch.Tensor):
             self.mark_static_input(value, guard=is_parameter_freezing())
             is_static_input = True
 
+        # Check to install params/bufs which are not input when requested
+        # Assumption: LocalSource is the only way our input is passed
+        # NOTE: These semantics currently will not install nested params/buffers, such as
+        # if a nn.Module is an input to a function, or list of parameters
+        if (
```

Comment:
Can probably tidy this up considerably, but these semantics are likely not exact just yet; cc @anijain2305 can this get special review to check these semantics first then we can tidy/consolidate with `make_graph_attr` below?

Reply:
Look for `UnspecializedParamBufferSource`. I think you only want to make those tensors a graph attribute whose sources have `UnspecializedParamBufferSource` in them.

Reply:
Yep restricting the third check in this conditional to just the check against `UnspecializedParamBufferSource` seems to do the trick :) I think I was overcomplicating this logic to try and keep some semantics in the non-export case which were not necessary

Reply:
In particular I was trying to keep semantics to not inline when nets are parameters such as:
```
        net = nn.Linear()
        def test_fn(x, net):
            return net(x)

        # When nn is in input, we don't install the params
        self.check_num_inputs_and_equality(test_fn, 3, 3, (x, net))
```

However, net can't be exported like this so we can choose whichever semantics we prefer here and have this be 
```
        self.check_num_inputs_and_equality(test_fn, 3, 1, (x, net))
```

Reply:
For this case, we can choose whatever sematics we prefer here, yeah.

Reply:
I think the only condition you need here is 

```
if config.install_params_as_graph_attr and is_from_unspecialized_param_buffer_source(source):
```

If not, can you show me example why?